src/ecdh.rs

```diff
@@ -135,52 +135,52 @@ impl SharedSecret {
         ss.set_len(32); // The default hash function is SHA256, which is 32 bytes long.
         ss
     }
+}
 
-
-    /// Creates a new shared secret from a pubkey and secret key with applied custom hash function.
-    /// The custom hash function must be in the form of `fn(x: [u8;32], y: [u8;32]) -> SharedSecret`
-    /// `SharedSecret` can be easily created via the `From` impl from arrays.
-    /// # Examples
-    /// ```
-    /// # #[cfg(any(feature = "alloc", features = "std"))] {
-    /// # use secp256k1::ecdh::SharedSecret;
-    /// # use secp256k1::{Secp256k1, PublicKey, SecretKey};
-    /// # fn sha2(_a: &[u8], _b: &[u8]) -> [u8; 32] {[0u8; 32]}
-    /// # let secp = Secp256k1::signing_only();
-    /// # let secret_key = SecretKey::from_slice(&[3u8; 32]).unwrap();
-    /// # let secret_key2 = SecretKey::from_slice(&[7u8; 32]).unwrap();
-    /// # let public_key = PublicKey::from_secret_key(&secp, &secret_key2);
-    ///
-    /// let secret = SharedSecret::new_with_hash(&public_key, &secret_key, |x,y| {
-    ///     let hash: [u8; 32] = sha2(&x,&y);
-    ///     hash.into()
-    /// });
-    /// # }
-    /// ```
-    pub fn new_with_hash<F>(point: &PublicKey, scalar: &SecretKey, mut hash_function: F) -> SharedSecret
-        where F: FnMut([u8; 32], [u8; 32]) -> SharedSecret {
-        let mut xy = [0u8; 64];
-
-        let res = unsafe {
-            ffi::secp256k1_ecdh(
-                ffi::secp256k1_context_no_precomp,
-                xy.as_mut_ptr(),
-                point.as_ptr(),
-                scalar.as_ptr(),
-                Some(c_callback),
-                ptr::null_mut(),
-            )
-        };
-        // Our callback *always* returns 1.
-        // and the scalar was verified to be valid(0 > scalar > group_order) via the type system
-        debug_assert_eq!(res, 1);
-
-        let mut x = [0u8; 32];
-        let mut y = [0u8; 32];
-        x.copy_from_slice(&xy[..32]);
-        y.copy_from_slice(&xy[32..]);
-        hash_function(x, y)
-    }
+/// Creates a shared point from public key and secret key.
+///
+/// Can be used like `SharedSecret` but caller is responsible for then hashing the returned buffer.
+/// This allows for the use of a custom hash function since `SharedSecret` uses SHA256.
```

Comment:
I think the wording should be stronger and with emphasis. Something like:

> **Important: use of strong cryptographic hash function may be critical to security! Do NOT use unless you understand cryptographical implications.** If not, use `SharedSecret` instead.

Reply:
Cheers, will use.